src/python/pants/backend/python/rules/run_setup_py.py

```diff
@@ -390,6 +464,29 @@ async def run_setup_py(
     return RunSetupPyResult(output_digest)
 
 
+@rule
+async def determine_setup_kwargs(
+    exported_target: ExportedTarget, union_membership: UnionMembership
+) -> SetupKwargs:
+    target = exported_target.target
+    plugin_requests = union_membership.get(SetupKwargsPluginRequest)  # type: ignore[misc]
+
+    # If no plugins, simply return what the user explicitly specified in a BUILD file.
+    if not plugin_requests:
+        return SetupKwargs(exported_target.provides.kwargs, address=target.address)
+
+    if len(plugin_requests) > 1:
+        possible_plugins = sorted(plugin.__name__ for plugin in plugin_requests)
+        raise ValueError(
+            f"Multiple of the registered `SetupKwargsPluginRequest`s can work on the target "
+            f"{target.address}, and it's ambiguous which to use: {possible_plugins}\n\nPlease "
+            "activate fewer implementations, or make the classmethod `is_valid()` more precise so "
```

Comment:
I don't actually see where `is_valid()` is used.

Reply:
Oh, that's because I forgot to wire it up. Oops.